recipes/categorical_anomaly.py

```diff
@@ -123,6 +139,26 @@ def tune_params(self,df, metrics_column):
         print("\nTuned LOF Parameters : {}".format(self.tuned_params))
         return(self.tuned_params)
 
+    def raster_to_vector(self,):
+        """
+        Given a filepath to a geotif file, will match the first layer of the raster tif to the
+        vector geometries located in the geodataframe given by the user.
+        Ex:
+        https://ghsl.jrc.ec.europa.eu/download.php?ds=pop
+        Using this dataset we can match the population geotif data with the FSA geometries in a geocohort,
+        and generate the population for each FSA
+        """
+        src = rasterio.open(self.filepath)
+        region = src.read(1, window=src.window(*self.df.geometry.total_bounds))
```

Comment:
prefer the use of context manager whenever available, so in case of exception the I/O connection is always handled for you:

```python
with rasterio.open(fpath) as src:
    ...
```